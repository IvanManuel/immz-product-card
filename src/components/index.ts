export { ProductImage } from './ProductImage';
export { ProductsButtons } from './ProductsButtons';
export { ProductTitle } from './ProductTitle';

import { ProductCardHOCProps } from '../interfaces/interfaces';
import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductImage } from './ProductImage';
import { ProductsButtons } from './ProductsButtons';
import { ProductTitle } from './ProductTitle';

export const ProductCard: ProductCardHOCProps = Object.assign( ProductCardHOC, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductsButtons,
})

export default ProductCard;