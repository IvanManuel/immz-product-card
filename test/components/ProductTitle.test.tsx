import React from "react";
import renderer from "react-test-renderer"
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {

    test('debe mostrar correctamente el componente con el tÃ­tulo personalizado', () => { 

        const wrapper = renderer.create(
            <ProductTitle title="Custom Product" />
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();

     });

     test('debe mostrar el compornente con el nombre del producto', () => { 

        const wrapper = renderer.create(
            <ProductCard product={product1 }>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();

     });

})
