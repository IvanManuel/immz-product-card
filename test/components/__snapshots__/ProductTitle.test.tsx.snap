// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe mostrar correctamente el componente con el t√≠tulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`ProductTitle debe mostrar el compornente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug
  </span>
</div>
`;
